Buildings/Applications/DHC/EnergyTransferStations/CoolingDirectControlledReturn.mo

```diff
@@ -0,0 +1,396 @@
+      tab="Controller", enable=
+        controllerType == Modelica.Blocks.Types.SimpleController.PI or
+        controllerType == Modelica.Blocks.Types.SimpleController.PID));
+
+  parameter Real xd_start=0
+    "Initial or guess value for state of derivative block"
+    annotation (Dialog(group="Initialization",
+      tab="Controller", enable=
+        controllerType == Modelica.Blocks.Types.SimpleController.PD or
+        controllerType == Modelica.Blocks.Types.SimpleController.PID));
+
+  parameter Real yCon_start=0
+    "Initial value of output from the controller"
+    annotation(Dialog(group="Initialization",
+      tab="Controller",
+      enable=initType == Modelica.Blocks.Types.InitPID.InitialOutput));
+
+  parameter Boolean reverseAction = true
+    "Set to true for throttling the water flow rate through a cooling 
+      coil controller"
+    annotation(Dialog(tab="Controller"));
+
+  // Advanced parameters
+  parameter Modelica.Fluid.Types.Dynamics energyDynamics=
+    Modelica.Fluid.Types.Dynamics.FixedInitial
+    "Type of energy balance: dynamic (3 initialization options) or steady state"
+    annotation(Dialog(tab="Advanced"));
+
+  parameter Modelica.SIunits.PressureDifference[3] dp_nominal=500*{1,-1,1}
+    "Nominal pressure drop in pipe junctions"
+    annotation(Dialog(tab="Advanced"));
+
+  Modelica.Blocks.Interfaces.RealInput TSetDisRet
+    "Setpoint for the minimum district return temperature"
+    annotation (Placement(transformation(extent={{-140,-140},{-100,-100}})));
+
+  Modelica.Blocks.Interfaces.RealOutput Q_flow(
+    final quantity="Power",
+    final unit="W",
+    displayUnit="kW")
+    "Measured power demand at the ETS"
+    annotation (Placement(transformation(extent={{100,140},{120,160}})));
+
+  Modelica.Blocks.Interfaces.RealOutput Q(
+    final quantity="Energy",
+    final unit="J",
+    displayUnit="kWh")
+    "Measured energy consumption at the ETS"
+    annotation (Placement(transformation(extent={{100,100},{120,120}})));
+
+  Buildings.Fluid.Actuators.Valves.TwoWayEqualPercentage conVal(
+    redeclare final package Medium = Medium,
+    final m_flow_nominal=mDis_flow_nominal,
+    final dpValve_nominal=dpConVal_nominal,
+    riseTime(displayUnit="s") = 60) "Control valve"
+    annotation (Placement(transformation(extent={{20,-50},{0,-70}})));
+
+  Buildings.Fluid.FixedResistances.Junction jun(
+    redeclare final package Medium = Medium,
+    final energyDynamics=energyDynamics,
+    final m_flow_nominal={mDis_flow_nominal,-mBui_flow_nominal,mByp_flow_nominal},
+    final dp_nominal=dp_nominal)
+    "Bypass junction"
+    annotation (Placement(transformation(extent={{30,50},{50,70}})));
+
+  Buildings.Fluid.Sensors.MassFlowRate senMasFlo(
+    redeclare final package Medium = Medium)
+    "District supply mass flow rate sensor"
+    annotation (Placement(transformation(extent={{-50,50},{-30,70}})));
+
+  Buildings.Fluid.Sensors.TemperatureTwoPort senTDisSup(
```

Comment:
The suggested changes have been incorporated. Thanks!

Reply:
Sorry I need to correct my suggestion. After discussion with the Project 1, setting `tau = if allowFlowReversal then 1 else 0` cannot be recommended as a general rule. The intent here is to reduce the number of state variables to reduce the number of iterations. However, the temperature sensor is often used in control loops where you would have direct feed-through of the variable in case you do not use a state. You end up with an algebraic loop that is likely to put more load on the solver than an additional state.